src/components/Navigation/PageMenu/PageMenuItem.tsx

```diff
@@ -0,0 +1,55 @@
+import { FC, useMemo } from 'react';
+
+import { Button, Container } from './PageMenuItem.styles';
+import { PageMenuItemType, usePageMenu } from 'src/providers/PageMenuProvider';
+
+interface PageMenuItemProps extends PageMenuItemType {
+  onlyShowSelectedChildren: boolean;
+  layer?: number;
+}
+
+const PageMenuItem: FC<PageMenuItemProps> = ({
+  label,
+  value,
+  children,
+  onlyShowSelectedChildren,
+  layer = 0,
+}) => {
+  const { isActive, setSelected } = usePageMenu();
+
+  const active = useMemo(
+    () => isActive({ label, value, children }),
+    [children, isActive, label, value]
+  );
+
+  const handleOnClick = () => {
+    if (!active) setSelected(value);
+  };
+
+  if (children) {
+    return (
+      <Container $layer={layer}>
+        <Button $active={active} onClick={handleOnClick} disabled={active}>
+          {label}
+        </Button>
+        {((onlyShowSelectedChildren && active) || !onlyShowSelectedChildren) &&
```

Comment:
Wont the PageMenu break at some point if you have no limit(that i have found 👀 ) on generations of children. For example great, great grand children :P

Reply:
I guess it would theoretically break if you have like tens of layers deep items, should we have some check in the pagemenuprovider that checks the depth?

Reply:
Wont the width be too small to write after like 4 depth? I am expecting each child to be indented from its parent like the children in the video you sent is.

Reply:
I think regarding a limit, it should come across in the types if we have one, and that might be a hassle for such a edge case? 🤔

Reply:
The width of PageMenu follows the width of the parent container 😄

Reply:
Fair enough <: 
I agree that we can keep it like this then, and we have full flexibility and freedom when using it.
I estimate 1-3 layers is the use case 98.54% of the time anyway 🧠